### ISSUE TITLE
Need simple method to get single int value from existing IEnumerable<MyFlagEnum>.

### ISSUE THREAD

**Issue by VictorioBerra** [ardalis/SmartEnum#409]
Super easy to go from an int representing multiple flags to an enum:

`IEnumerable<MySmartFlagEnum> example = MySmartFlagEnum.FromValue(3); // Represents flag 1 | 2`

Not as clean going back?

`example.Sum(x => x.Value)`

Proposed:

`int origNumber = MySmartFlagEnum.FromValue(example)`

Sorry if there is something similar that already exists.

**Comment by ardalis** [comment]
You're saying `example.Sum(x => x.Value)` doesn't work? What does it return?

**Comment by VictorioBerra** [comment]
@ardalis Sum works. It just took me a lot of fiddling and reading the docs to realize I had to Sum them to get them back and could not just use a built in method like FromValue().

Is there any way to simply have an overload for `FromValue()` that takes a SmartFlagEnum and does the Sum? This way, next time Intellisense will be enough to give me the answer instead of being confused why theres no helpful method to go both ways.

**Comment by ardalis** [comment]
You have an IEnumerable, though, so any method or overload would need to be an extension method on IEnumerable, right?

@sdepouw thoughts on this, since you did the Flags implementation? Am I missing something?

**Comment by sdepouw** [comment]
I didn't write the original `SmartFlagEnum` implementation myself but I did poke around a bit to try this suggestion out. `mySmartEnumValues.Sum(v => v.Value)` may be your best bet in the short term at least @VictorioBerra , based off a little experimenting I did. Since `TValue` is generic, it's not as easy to supply a `FromValue()` overload that takes in enumerable of `TSmartFlagEnum` and gives back `TValue`, since generic math would be required:

![image](https://github.com/ardalis/SmartEnum/assets/4809334/8ca3d799-c7fa-4e3f-a504-e9822d161d5b)

That is somewhat possible in later versions of .NET (.NET 7+ has `INumber<T>`), but .NET 6 (a version SmartEnum builds under) wouldn't be able to use it. We'd also have to retrofit `SmartFlagEnum` and related classes to properly constrain `TValue` to support the generic math to boot (I started trying that but it was leading a larger and larger changeset). One could implement a different `FromValue()` overload for each type of number that `TValue` could be but that may end up being too cumbersome to maintain.
